### ISSUE TITLE
Copying VMs doesn't reset /etc/machine-id, causing duplicate IPv4 IPs.

### ISSUE THREAD

**Issue by KellenRenshaw** [lxc/lxd#8717]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 20.04
 * The output of "lxc info":
[host_4.0.5_lxc-info.txt](https://github.com/lxc/lxd/files/6386486/host_4.0.5_lxc-info.txt)
[host_4.13_lxc-info.txt](https://github.com/lxc/lxd/files/6386488/host_4.13_lxc-info.txt)

# Issue description

Copying a Focal VM (containers appear to be unaffected) results in the duplicated VM getting the same IPv4 address via DHCP. This happens both with LXD's in-built dnsmasq service and a third-party DHCP solution. The MAC addresses on the copy are different from the original.

Tracked down to the machine-id in /etc/machine-id being the same in the copies/clones. Resetting the machine-id of the copy with `echo -n > /etc/machine-id` and rebooting results in the expected DHCP behavior of receiving a different IPv4 address relative to the parent VM.

Example with "testapp" being the parent, "clone" showing the issue, and "testapp-clone" showing the result of the workaround.

```
+---------------+---------+-----------------------+------------------------------------------------+-----------------+-----------+
|     NAME      |  STATE  |         IPV4          |                      IPV6                      |      TYPE       | SNAPSHOTS |
+---------------+---------+-----------------------+------------------------------------------------+-----------------+-----------+
| clone         | RUNNING | 10.14.70.198 (enp5s0) | fd42:e7bf:4ff:10bb:216:3eff:fe3a:b09e (enp5s0) | VIRTUAL-MACHINE | 0         |
+---------------+---------+-----------------------+------------------------------------------------+-----------------+-----------+
| testapp       | RUNNING | 10.14.70.198 (enp5s0) | fd42:e7bf:4ff:10bb:216:3eff:fe84:a49b (enp5s0) | VIRTUAL-MACHINE | 0         |
+---------------+---------+-----------------------+------------------------------------------------+-----------------+-----------+
| testapp-clone | RUNNING | 10.14.70.45 (enp5s0)  | fd42:e7bf:4ff:10bb:216:3eff:feeb:1e1 (enp5s0)  | VIRTUAL-MACHINE | 0         |
+---------------+---------+-----------------------+------------------------------------------------+-----------------+-----------+


Notes: testapp-clone has had the workaround of "echo -n > /etc/machine-id ; reboot" applied to it.
```

# Steps to reproduce

 1. Obtain fresh Ubuntu Focal host, install LXD, tested with 4.0.5 and 4.13.
 2. Run `lxd init`, taking the defaults for all choices.
 3. Launch Focal VM with `lxc launch ubuntu:focal --vm test1`.
 4. Shutdown the test1 VM and copy it to "test2" with `lxc stop test1; lxc copy test1 test2`
 5. Start both VMs.
 6. They will end up with identical IPv4 addresses.

# Information to attach

 - [ ] Any relevant kernel output (`dmesg`)
 - [ ] Container log (`lxc info NAME --show-log`)
 - [ ] Container configuration (`lxc config show NAME --expanded`)
 - [X ] Main daemon log (at /var/log/lxd/lxd.log or /var/snap/lxd/common/lxd/logs/lxd.log) for 4.0.5 
[lxd_4.0.5_log.txt](https://github.com/lxc/lxd/files/6386554/lxd_4.0.5_log.txt)
 - [ ] Output of the client with --debug
 - [ ] Output of the daemon with --debug (alternatively output of `lxc monitor` while reproducing the issue)

**Comment by stgraber** [comment]
Hmm, normally that's what cloud-init is supposed to be doing for you.

**Comment by stgraber** [comment]
I've confirmed that on our side we're doing the right thing by cleaning volatile.uuid which then causes a different cloud-init id from being used, this re-triggers cloud-init (can be confirmed by the VM hostname being correct), but cloud-init doesn't seem to care about re-generating machine-id causing this issue.

There's a very old bug here https://bugs.launchpad.net/ubuntu/+source/cloud-init/+bug/1563951 which appears to still need fixing...

Closing as LXD relies on cloud-init for such handling and cloud-init doesn't appear to be re-generating machine-id when creating an instance from another one. I suspect the same should be reproducible in the cloud too.
